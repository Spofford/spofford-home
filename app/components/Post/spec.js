import React from "react"
import expect from "expect"
import { shallow } from "enzyme"

import { Post } from "./"

describe("<Post />", () => {
  it("should render", () => {
    const renderedComponent = shallow(
      <Post />
    )
    expect(renderedComponent.is("div")).toEqual(true)
  })
})
