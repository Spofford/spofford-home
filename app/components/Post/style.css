.post h2, .post h3 {
  margin-bottom:15px;
}

.post p {
  line-height:1.8em;
  margin-bottom:10px;
}

.post blockquote {
  font-size:1.5em;
  font-style: italic;
  padding-left:15px;
}

.post blockquote a {
  text-transform: none;
}

.post img {
  width:100%;
}

.post .fa {
  margin-right:10px;
}
