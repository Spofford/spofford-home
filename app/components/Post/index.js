import React from "react"
import * as contentful from 'contentful'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import { BrowserRouter as Router, Route, Link} from 'react-router-dom'
import FontAwesome from "react-fontawesome";

import { default as Header } from "../Header"

export class Post extends React.Component {
  state = {
    model: {}
  }

  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    console.log(this.props.match.params.entity)
    this.fetchModel().then(this.setModel);
  }

  fetchModel = () => this.client.getEntry(this.props.match.params.entity)

  setModel = response => {
    this.setState({
      model: response.fields
    })
  }

  render() {

    return (
      <div className="post">
        <Header />
        <div className="body-container">
          <div className="copy-container">
            <h2>{this.state.model.head}</h2>
            <h3>{this.state.model.subhead}</h3>
            <ReactMarkdown source={this.state.model.bodyText} />
            <Link to='/' className="return-home"><FontAwesome name='chevron-left' />Return to home</Link>
          </div>
        </div>
      </div>
    )
  }
}

export default Post
