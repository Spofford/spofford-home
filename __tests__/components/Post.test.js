import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'
import * as contentful from 'contentful'

import Post from '../../app/components/Post';

const props = {
  match: {
    params: {
      entity: "719Ed2edIk8Yi2Img602A"
    }
  }
}


describe('<Post />', () => {
  describe('render()', () => {
    beforeEach(() => {
      var client = contentful.createClient({
        space: 'cahjy08ew1qz',
        accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
      });
    });

    test('renders the component', async () => {
      const wrapper = shallow(<Post {...props} />);
      const instance = wrapper.instance();

      const data = await instance.fetchModel();
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('calls setModel on mount', async () => {
      const wrapper = shallow(<Post {...props} />);
      const spy = jest.spyOn(wrapper.instance(), "setModel");
      wrapper.instance().forceUpdate();
      wrapper.instance().componentDidMount();

      const data = await wrapper.instance().fetchModel();
      expect(spy).toHaveBeenCalled();
    })

    // Test for contentful content

  });
});
