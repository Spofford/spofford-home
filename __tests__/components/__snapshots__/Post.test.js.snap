// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> render() renders the component 1`] = `
<div
  className="post"
>
  <Connect(Header) />
  <div
    className="body-container"
  >
    <div
      className="copy-container"
    >
      <h2 />
      <h3 />
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        plugins={Array []}
        rawSourcePos={false}
        renderers={Object {}}
        skipHtml={false}
        sourcePos={false}
        transformLinkUri={[Function]}
      />
      <Link
        className="return-home"
        replace={false}
        to="/"
      >
        <FontAwesome
          name="chevron-left"
        />
        Return to home
      </Link>
    </div>
  </div>
</div>
`;
